nightly/index.html

```diff
@@ -0,0 +1,215 @@
+              $(this).datepicker('hideWidget');
+              console.log(new_url);
+              loadJSON(new_url);
+            })
+          });
+      function loadJSON(url) {
+        var xhttp;
+        xhttp=new XMLHttpRequest();
+        xhttp.onreadystatechange = function() {
+          if (this.readyState == 4 && this.status == 200) {
+            jsonText = xhttp.responseText
+            buildPage(jsonText, false);
+          } else {
+            $('#target').html('No builds available for selected date.');
+          }
+        };
+        xhttp.open("GET", url, true);
+        xhttp.send();
+      }
+      function buildPage(jsonText, is_rebuild) {
+        const corretto_domain = 'https://corretto.aws';
+        var data = JSON.parse(jsonText);
+        $('#datepicker').datepicker('update', data.date);
+
+        const select_version = document.getElementById('versionSelect');
+        const select_platform = document.getElementById('platformSelect');
+        const select_architecture = document.getElementById('architectureSelect');
+
+        const known_versions = new Set(['All Versions'])
+        const known_platforms = new Set(['All Platforms'])
+        const known_architectures = new Set(['All Architectures'])
+        $html = '';
+        $html += '<table class="table table-bordered table-striped">'
+        $html += '<thead>'
+        $html += '<tr>'
+        $html += '<th scope="col">Version</th>'
+        $html += '<th scope="col">Branch</th>'
+        $html += '<th scope="col">Commit</th>'
+        $html += '<th scope="col">Debug Level</th>'
+        $html += '<th scope="col">Platform</th>'
+        $html += '<th scope="col">Architecture</th>'
+        $html += '<th scope="col">Artifacts</th>'
+        $html += '</tr></thead>'
+        $html += '<tbody>'
+        for (const version of Object.keys(data.versions).sort(correttoCompare)) {
+          known_versions.add(version);
+          for (branch in data.versions[version].branches) {
+            for (commit in data.versions[version].branches[branch].commits) {
+              for (debug_level in data.versions[version].branches[branch].commits[commit].debug_levels) {
+                for (platform in data.versions[version].branches[branch].commits[commit].debug_levels[debug_level].platforms) {
+                  known_platforms.add(platform);
+                  for (arch in data.versions[version].branches[branch].commits[commit].debug_levels[debug_level].platforms[platform].architectures) {
+                    architecture = data.versions[version].branches[branch].commits[commit].debug_levels[debug_level].platforms[platform].architectures[arch];
+                    known_architectures.add(arch);
+                    if (select_version.selectedIndex != 0 && select_version.options[select_version.selectedIndex].value != version) {
```

Comment:
Why do we skip when its index 0?

Can we put the first two if's early in the nested fors?

Reply:
I'll add a comment, index 0 is the "All Versions" entry. I choose index checking rather than text to make it easier for me to change it.